crates/rpc/rpc/src/eth/helpers/signer.rs

```diff
@@ -79,18 +82,28 @@ impl EthSigner for DevSigner {
         self.sign_hash(hash, address)
     }
 
-    fn sign_transaction(
+    async fn sign_transaction(
         &self,
-        request: TypedTransactionRequest,
+        request: TransactionRequest,
         address: &Address,
     ) -> Result<TransactionSigned> {
-        // convert to primitive transaction
-        let transaction =
-            to_primitive_transaction(request).ok_or(SignError::InvalidTransactionRequest)?;
-        let tx_signature_hash = transaction.signature_hash();
-        let signature = self.sign_hash(tx_signature_hash, *address)?;
+        // create local signer wallet from signing key
+        let sk_bytes = self.accounts.get(address).ok_or(SignError::NoAccount)?.secret_bytes();
+        let wallet =
+            PrivateKeySigner::from_bytes(&sk_bytes.into()).map_err(|_| SignError::NoAccount)?;
+        let signer = EthereumWallet::from(wallet);
```

Comment:
perhaps we can simplify this by changing SecretKey to

`PrivateKeySigner`?

https://github.com/paradigmxyz/reth/blob/5a2d892579dda42d3b2ab82836a7b8a38d8ec321/crates/rpc/rpc/src/eth/helpers/signer.rs#L32-L32

https://github.com/alloy-rs/alloy/blob/8b57b52f8c796ce785e4530bf9e6fc73ea743ecc/crates/signer-local/src/lib.rs#L36-L37

Reply:
updated now using signer api